### ISSUE TITLE
Bug: Issue with peer dependencies

### ISSUE THREAD

**Issue by jerome-benoit** [azat-io/eslint-plugin-perfectionist#378]
### Describe the bug

Running `pnpm i -D eslint-plugin-perfectionist` triggers: 

```
 WARN  Issues with peer dependencies found
.
├─┬ eslint-plugin-import 2.30.0
│ └── ✕ unmet peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8": found 9.15.0
└─┬ eslint-plugin-perfectionist 4.0.0
  └─┬ @azat-io/eslint-config 2.1.2
    ├─┬ eslint-plugin-sonarjs 2.0.4
    │ ├─┬ @typescript-eslint/utils 7.16.1
    │ │ └── ✕ unmet peer eslint@^8.56.0: found 9.15.0
    │ ├─┬ eslint-plugin-import 2.30.0
    │ │ └── ✕ unmet peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8": found 9.15.0
    │ ├─┬ eslint-plugin-react-hooks 4.6.2
    │ │ └── ✕ unmet peer eslint@"^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0 || ^7.0.0 || ^8.0.0-0": found 9.15.0
    │ └─┬ @typescript-eslint/eslint-plugin 7.16.1
    │   ├── ✕ unmet peer eslint@^8.56.0: found 9.15.0
    │   ├── ✕ unmet peer @typescript-eslint/parser@^7.0.0: found 8.15.0
    │   └─┬ @typescript-eslint/type-utils 7.16.1
    │     └── ✕ unmet peer eslint@^8.56.0: found 9.15.0
    ├─┬ eslint-plugin-qwik 1.10.0
    │ └── ✕ unmet peer eslint@^8.57.0: found 9.15.0
    └─┬ eslint-plugin-perfectionist 3.9.1
      └── ✕ unmet peer svelte-eslint-parser@^0.41.1: found 0.43.0
```

`eslint-plugin-sonarjs` is not maintained: https://github.com/SonarSource/eslint-plugin-sonarjs

### Code example

Not relevant

### ESLint version

9.15.0

### ESLint Plugin Perfectionist version

4.0.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by lishaduck** [comment]
For reference, v3 is 38MB[^1], while v4 is 144MB[^1]. @azat-io/eslint-config is 143MB, so that should fix it 😂 (there's certainly some shared deps, but the point stands).

Also: eslint-plugin-sonarjs is maintained, it's just in a new repo.

[^1]: 5MB without peers
[^2]: 112 without peers
[^3]: 110 without peers

**Comment by azat-io** [comment]
Oops. Fixed in 191902a.
Released in [v4.0.1](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.0.1).

**Comment by lishaduck** [comment]
v4.0.1 is now 37M, 4.8MB without peers! 🥳
